dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -376,6 +383,72 @@ class PerfTest {
   }
 }
 
+class E2EPerfTest extends PerfTest {
+  const E2EPerfTest(
+    String testDirectory,
+    String testTarget,
+  ) : super(
+    testDirectory,
+    testTarget,
+    'e2e',
+    testDriver: 'test_driver/e2e_test.dart',
+  );
+
+  @override
+  Future<TaskResult> internalRun({
```

Comment:
I'm not completely sure why this `internalRun` needs to be rewritten... what lines of code should I focus on reading?

Reply:
L439 ~ 446, the key list. This E2E version cannot get vsync related metrics.

Reply:
Other than that, I'm expecting E2E to share `testDriver` and to have different timeline file name. It doesn't make sense to still name the file `**.timeline_summary.json` since its of different format.

Reply:
Mentioning this, I think '--trace-startup' is no longer needed. I'll delay the update to keep the above line numbers valid.

Reply:
To minimize duplicate code, maybe we can let `internalRun` take options like `bool reportsVsyncMetrics = true, bool needsTraceStartup = true`, and then just use `internalRun(reportsVsyncMetrics: false, needsTraceStartup: false)` in the e2e test?

Reply:
@CareF : how do you think about the last comment about reducing duplicate code?

Reply:
I had a slightly different design but still reduce the duplicate. Let me know if that looks good.